siege_utilities/geo/spatial_data.py

```
@@ -624,28 +652,44 @@ def _construct_filename_with_fips_validation(self, year: int, boundary_type: str
+                    year, max_known, congress_num,
+                )
+            # Census dropped the national CD file after 2021; 2022+ only publishes 56 per-state files.
+            if year >= 2022 and not state_fips:
                 raise BoundaryConfigurationError(
-                    f"Congressional district boundary type '{boundary_type}' requires a "
-                    f"congress number. Provide it via the boundary_type name (e.g., 'cd118') "
-                    f"or the congress_number parameter.",
+                    f"Congressional district national file is unavailable for {year}+. "
+                    f"Provide state_fips to download the per-state file.",
                     context={"boundary_type": boundary_type, "year": year},
                 )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)